packages/react/src/Testimonial/Testimonial.test.tsx

```diff
@@ -0,0 +1,515 @@
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>{mockName}</Testimonial.Name>
+      </Testimonial>,
+    )
+
+    const name = getByText(mockName)
+    const position = queryByText(mockPosition)
+    expect(name).toBeInTheDocument()
+    expect(position).not.toBeInTheDocument()
+  })
+
+  it('renders name with position', () => {
+    const {getByText} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name position={mockPosition}>{mockName}</Testimonial.Name>
+      </Testimonial>,
+    )
+
+    const name = getByText(mockName)
+    const position = getByText(mockPosition)
+    expect(name).toBeInTheDocument()
+    expect(position).toBeInTheDocument()
+  })
+
+  it('applies custom className to name', () => {
+    const {getByText} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name className="custom-name">{mockName}</Testimonial.Name>
+      </Testimonial>,
+    )
+
+    const figcaption = getByText(mockName).closest('figcaption')
+    expect(figcaption).toHaveClass('custom-name')
+  })
+
+  it('supports RefObject for Testimonial.Name component', () => {
+    const refObject = React.createRef<HTMLElement>()
+    const {getByText} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name ref={refObject}>{mockName}</Testimonial.Name>
+      </Testimonial>,
+    )
+
+    const figcaption = getByText(mockName).closest('figcaption')
+    expect(refObject.current).toBe(figcaption)
+  })
+
+  it('supports functional ref for Testimonial.Name component', () => {
+    const mockRef = jest.fn()
+    const {getByText} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name ref={mockRef}>{mockName}</Testimonial.Name>
+      </Testimonial>,
+    )
+
+    const figcaption = getByText(mockName).closest('figcaption')
+    expect(mockRef).toHaveBeenCalledWith(figcaption)
+  })
+
+  it('renders avatar image', () => {
+    const {getByRole} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+        <Testimonial.Avatar src={mockAvatarSrc} alt={mockAvatarAlt} />
+      </Testimonial>,
+    )
+
+    const avatar = getByRole('img', {name: mockAvatarAlt})
+    expect(avatar).toBeInTheDocument()
+    expect(avatar).toHaveAttribute('src', mockAvatarSrc)
+  })
+
+  it('uses fixed avatar size of 48px', () => {
+    const {getByAltText} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+        <Testimonial.Avatar src={mockAvatarSrc} alt={mockAvatarAlt} size={64} />
+      </Testimonial>,
+    )
+
+    const avatarContainer = getByAltText(mockAvatarAlt)
+    expect(avatarContainer).toBeInTheDocument()
+  })
```

Comment:
I think the test name here is wrong? Prop value is `64` but test name is `48`?. I might be missing something? 🤔. Should this be asserting for a class or specific dimension property too rather than checking it's in the document?

Reply:
The test name could be clearer here. I suppose what I'm actually testing is that the size of the `Avatar` _can't_ be overridden as it's intentionally fixed to 48px. Let me make that a bit clearer, and improve the assertion.

Reply:
Improved in 412e52de200b7633f1db7f535451c46bcde24a6a, lmk what you think